cmd/cosign/cli/attach/attestation.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2021 The Sigstore Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package attach
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/pkg/errors"
+	ssldsse "github.com/secure-systems-lab/go-securesystemslib/dsse"
+	"github.com/sigstore/cosign/cmd/cosign/cli/fulcio"
+	"github.com/sigstore/cosign/cmd/cosign/cli/options"
+	"github.com/sigstore/cosign/cmd/cosign/cli/rekor"
+	"github.com/sigstore/cosign/pkg/cosign"
+	"github.com/sigstore/cosign/pkg/cosign/pivkey"
+	"github.com/sigstore/cosign/pkg/cosign/pkcs11key"
+	"github.com/sigstore/cosign/pkg/oci/mutate"
+	ociremote "github.com/sigstore/cosign/pkg/oci/remote"
+	"github.com/sigstore/cosign/pkg/oci/static"
+	sigs "github.com/sigstore/cosign/pkg/signature"
+	"github.com/sigstore/cosign/pkg/types"
+	"github.com/sigstore/sigstore/pkg/signature/dsse"
+)
+
+func AttestationCmd(ctx context.Context, regOpts options.RegistryOptions, keyRef string, sk options.SecurityKeyOptions, signedPayload, rekorURL, imageRef string) error {
+	ociremoteOpts, err := regOpts.ClientOpts(ctx)
+	if err != nil {
+		return errors.Wrap(err, "constructing client options")
+	}
+	co := &cosign.CheckOpts{
+		RegistryClientOpts: ociremoteOpts,
+	}
+
+	if options.EnableExperimental() {
+		if rekorURL != "" {
+			rekorClient, err := rekor.NewClient(rekorURL)
+			if err != nil {
+				return errors.Wrap(err, "creating Rekor client")
+			}
+			co.RekorClient = rekorClient
+		}
+		co.RootCerts = fulcio.GetRoots()
+	}
+
+	// Keys are optional!
+	if keyRef != "" {
```

Comment:
Sorry! I think I wasn't clear earlier. We don't actually need to verify the signature itself (that seems a bit overkill), just verify that the attestation object is well-formed and contains a signature.

My worry was someone doing something like: `cosign attach attestation --attestation=README.md <image>` and assuming it would sign the README.md file automatically.

Reply:
ah okay, now I got it, thank you, it should be fixed right now. 🤝